### PR TITLE
PIM-7394: add slash as an allowed character for the identifier of a product in the API

### PR DESCRIPTION
**Pull request by ahocquard**
**Description (for Contributor and Core Developer)**

Not really testable:
- fake request use a fake router
- ideally, it should test it with apache as it is infrastructure stuff

I tested and it works with GET and PATCH:
- /products//my_identifier  ("/my_identifier")
- /products/my_identifier/  ("/my_identifier/")
- /products/my_id/entifier  ("my_id/entifier")

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo?
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: CHANGELOG-1.7.md [hunk 1]
```diff
@@ -33,7 +34,6 @@
 ## Improvements
 
 - IM-824: Change message when the user or email is not valid to a more generic message
-- PIM-7253: Do not schedule mongo queries to update normalizedData for newly created entities
```
Comment (Doodoune) [inline CHANGELOG-1.7.md]: Why remove this line ?

Comment (pierallard) [reply CHANGELOG-1.7.md]: Rebase  ¯\_(ツ)_/¯ ?

File: src/Pim/Bundle/ApiBundle/Resources/config/routing/product.yml [hunk 2]
```diff
@@ -7,6 +7,8 @@ pim_api_product_get:
     path: /products/{code}
     defaults: { _controller: pim_api.controller.product:getAction, _format: json }
     methods: [GET]
+    requirements:
+        code: .+
```
Comment (pierallard) [inline src/Pim/Bundle/ApiBundle/Resources/config/routing/product.yml]: I don't get the solution... I dont get why "at lease one char" will solve it. Can you explain ?

File: CHANGELOG-1.7.md (hunk 1 above)
Comment (ahocquard) [reply CHANGELOG-1.7.md]: argh, I hate our way to handle this changelog !!!!

File: src/Pim/Bundle/ApiBundle/Resources/config/routing/product.yml (hunk 2 above)
Comment (ahocquard) [reply src/Pim/Bundle/ApiBundle/Resources/config/routing/product.yml]: https://symfony.com/doc/2.7/routing/slash_in_parameter.html

**Comment by ahocquard** [comment]
@Doodoune No need to add it in the documentation, as I said in the explanation, it works even with a slash at the beginning or at the end.

**Pull request merged**